### PR TITLE
added database sync plugin

### PR DESCRIPTION
**Pull request by mkurek**
added database sync plugin to get databases from ralph, which will allow to show them as components in components view and charge them for their usages of resources.

requires: https://github.com/allegro/ralph/pull/1274

### CODE REVIEW

**Comment by kula1922** [comment]
:ok:

**Comment by kula1922** [comment]
:ok:

File: src/ralph_scrooge/models/pricing_object.py [hunk 1]
```diff
@@ -424,3 +425,49 @@ class DailyVIPInfo(DailyPricingObject):
...
+                pricing_object=self,
+                database_info=self,
+                date=date,
+                service_environment=self.service_environment,
+            )
+
+
+class DailyDatabaseInfo(DailyPricingObject):
+    parent_device = db.ForeignKey(
+        DailyAssetInfo,
```
Comment (quamilek) [inline src/ralph_scrooge/models/pricing_object.py]: Why `DailyAssetInfo`?

File: src/ralph_scrooge/plugins/collect/database.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+
+def get_model(ralph_database):
+    model = PricingObjectModel.objects.get_or_create(
+        model_id=ralph_database['type_id'],
+        type_id=PRICING_OBJECT_TYPES.DATABASE,
+        defaults=dict(
+            name=ralph_database['type'],
+        )
+    )[0]
```
Comment (quamilek) [inline src/ralph_scrooge/plugins/collect/database.py]:
```
model, created = PricingObjectModel.objects.get_or_create(
    model_id=ralph_database['type_id'],
    type_id=PRICING_OBJECT_TYPES.DATABASE,
    defaults=dict(
        name=ralph_database['type'],
    )
)
```
looks better

File: src/ralph_scrooge/plugins/collect/database.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                database_type
+            )
+        except UnknownServiceEnvironmentNotConfiguredError:
+            msg = 'Unknown service environment not configured for {}'.format(
+                database_type
+            )
+            logger.error(msg)
+            return (False, msg)
+
+    for database_type in settings.DATABASE_TYPES:
```
Comment (quamilek) [inline src/ralph_scrooge/plugins/collect/database.py]: I think that you could use one `for` ?

File: src/ralph_scrooge/plugins/collect/database.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+)
+
+from ralph_scrooge.plugins.collect._exceptions import (
+    UnknownServiceEnvironmentNotConfiguredError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_model(ralph_database):
```
Comment (quamilek) [inline src/ralph_scrooge/plugins/collect/database.py]: Name should be more descriptive, is simillar to: 
https://github.com/allegro/ralph_pricing/pull/361/files#diff-1271648e9b72d66d1b1723e023b7938eR28

File: src/ralph_scrooge/plugins/collect/database.py (hunk 3 above)
Comment (kula1922) [inline src/ralph_scrooge/plugins/collect/database.py]: Its look like case where if there is no unknown_service_environment for any database then do nothing. Thats true?

File: src/ralph_scrooge/plugins/collect/database.py (hunk 2 above)
Comment (kula1922) [inline src/ralph_scrooge/plugins/collect/database.py]: looks better? why?

File: src/ralph_scrooge/plugins/collect/database.py (hunk 4 above)
Comment (kula1922) [inline src/ralph_scrooge/plugins/collect/database.py]: Collects plugins are similar and we know about that. We are going to separate some common methods (like get_model) but now there is no time. This method return model so get_model is a correct namespace for me.

File: src/ralph_scrooge/plugins/collect/database.py (hunk 3 above)
Comment (mkurek) [inline src/ralph_scrooge/plugins/collect/database.py]: Exactly - if unknown service environment is not set for all of the database types, we don't want to run any further processing.

**Comment by quamilek** [comment]
:ok:

File: src/ralph_scrooge/models/pricing_object.py [hunk 5]
```diff
@@ -424,3 +425,49 @@ class DailyVIPInfo(DailyPricingObject):
...
+        verbose_name=_("parent device"),
+        related_name='databases',
+        null=True,
+        blank=True,
+    )
+
+    class Meta:
+        app_label = 'ralph_scrooge'
+
+    def get_daily_pricing_object(self, date):
```
Comment (kula1922) [inline src/ralph_scrooge/models/pricing_object.py]: docstring

**Pull request merged**